core/src/main/java/de/danoeh/antennapod/core/cast/RemoteMedia.java

```diff
@@ -0,0 +1,358 @@
+                .setMetadata(metadata);
+        if (duration > 0) {
+            builder.setStreamDuration(duration);
+        }
+        return builder.build();
+    }
+
+    public String getEpisodeIdentifier() {
+        return itemIdentifier;
+    }
+
+    public String getFeedUrl() {
+        return feedUrl;
+    }
+
+    public FeedMedia lookForFeedMedia() {
+        FeedItem feedItem = DBReader.getFeedItem(feedUrl, itemIdentifier);
+        if (feedItem == null) {
+            return null;
+        }
+        return feedItem.getMedia();
+    }
+
+    @Override
+    public void writeToPreferences(SharedPreferences.Editor prefEditor) {
+        //it seems pointless to do it, since the session should be kept by the remote device.
+    }
+
+    @Override
+    public void loadMetadata() throws PlayableException {
+        //Already loaded
+    }
+
+    @Override
+    public void loadChapterMarks() {
+        ChapterUtils.loadChaptersFromStreamUrl(this);
+    }
+
+    @Override
+    public String getEpisodeTitle() {
+        return episodeTitle;
+    }
+
+    @Override
+    public List<Chapter> getChapters() {
+        return chapters;
+    }
+
+    @Override
+    public String getWebsiteLink() {
+        if (episodeLink != null) {
+            return episodeLink;
+        } else {
+            return feedUrl;
+        }
+    }
+
+    @Override
+    public String getPaymentLink() {
+        return null;
+    }
+
+    @Override
+    public String getFeedTitle() {
+        return feedTitle;
+    }
+
+    @Override
+    public Object getIdentifier() {
+        return itemIdentifier + "@" + feedUrl;
+    }
+
+    @Override
+    public int getDuration() {
+        return duration;
+    }
+
+    @Override
+    public int getPosition() {
+        return position;
+    }
+
+    @Override
+    public long getLastPlayedTime() {
+        return lastPlayedTime;
+    }
+
+    @Override
+    public MediaType getMediaType() {
+        if (TextUtils.isEmpty(mime_type)) {
+            return MediaType.UNKNOWN;
+        } else {
+            if (mime_type.startsWith("audio")) {
+                return MediaType.AUDIO;
+            } else if (mime_type.startsWith("video")) {
+                return MediaType.VIDEO;
+            } else if (mime_type.equals("application/ogg")) {
+                return MediaType.AUDIO;
+            }
+        }
+        return MediaType.UNKNOWN;
+    }
+
+    @Override
+    public String getLocalMediaUrl() {
+        return null;
+    }
+
+    @Override
+    public String getStreamUrl() {
+        return downloadUrl;
+    }
+
+    @Override
+    public boolean localFileAvailable() {
+        return false;
+    }
+
+    @Override
+    public boolean streamAvailable() {
+        return true;
+    }
+
+    @Override
+    public void saveCurrentPosition(SharedPreferences pref, int newPosition, long timestamp) {
+        //we're not saving playback information for this kind of items on preferences
```

Comment:
What I had in mind was that this format was only for media that was picked up from the chromecast, but somehow could not be matched to any local one, as we want to use FeedMedia by default.
In the current setting, this is probably useless because we're not allowing yet for instances of AP to share a chromecast session (unless somehow the user deletes the feed locally and the chromecast keeps playing it).
If there would be a purpose to saving it in the preferences then yes, we should implement it.